### PR TITLE
Add VM architecture diagram

### PR DESCRIPTION
**Pull request by howardjohn**
This adds a new page describing the architecture of VM deployments. This is requested by TOC as a blocker for beta promotion.

### CODE REVIEW

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:31]:
```suggestion
This ensures that the VMs have a stable address to connect to when they are bootstrapping. Pods and virtual machines can communicate directly with each other without requiring any intermediate Gateway.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:27]:
```suggestion
In this mesh, there is a single [network](/docs/ops/deployment/deployment-models/#network-models), implying that pods and virtual machines can communicate directly with each other.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:44]:
```suggestion
In this mesh, there are multiple [networks](/docs/ops/deployment/deployment-models/#network-models), implying that pods and virtual machines are not able to communicate directly with each other.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:62]:
```suggestion
Istio provides two new mechanisms to represent virtual machine workloads:
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:64]:
```suggestion
* [`WorkloadGroup`](/docs/reference/config/networking/workload-group/) represent a logical group of virtual machine workloads that share common properties. This is similar to a `Deployment` in Kubernetes.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:65]:
```suggestion
* [`WorkloadEntry`](/docs/reference/config/networking/workload-entry/) represent a single instance of a virtual machine workload. This is similar to a `Pod` in Kubernetes.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:67]:
```suggestion
Creating these resources (WorkloadGroup and WorkloadEntry) does not result in provisioning of any resources or running any virtual machine workloads.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:88]:
```suggestion
Once the virtual machine has been [configured and added to the mesh](/docs/setup/install/virtual-machine/#configure-the-virtual-machine), a corresponding `WorkloadEntry` will be automatically created by the Istio control plane.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:107]:
```suggestion
This `WorkloadEntry` resource describes a single instance of a workload, similar to a pod in Kubernetes. When the workload is removed from the mesh, the `WorkloadEntry` resource will
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:108]:
```suggestion
be automatically removed.  Additionally, if any probes are configured in the `WorkloadGroup` resource, the Istio control plane automatically updates the health status of associated `WorkloadEntry` instances. 
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:112]:
```suggestion
In order for consumers to reliably call your workload, it's recommended to declare a `Service` association. This allows clients to reach a stable hostname, like `product.default.svc.cluster.local`, rather than ephemeral IP addresses. This also enables you to use advanced routing capabilities in Istio via `DestinationRule` and `VirtualService` APIs.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:114]:
```suggestion
Any Kubernetes service transparently selects workloads across pods and virtual machines via the selector fields which are matched with pod and `WorkloadEntry` labels respectively.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:128]:
```suggestion
Kubernetes provides DNS resolution in pods for `Service` names  allowing pods to easily communicate with one another by stable hostnames.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:131]:
```suggestion
For virtual machine expansion Istio provides similar functionality via [DNS Proxy](/docs/ops/configuration/traffic-management/dns-proxy/).
This feature redirects all DNS queries from the virtual machine workload to the Istio proxy, which maintains a mapping of hostnames to IP addresses.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:133]:
```suggestion
As a result, workloads running on virtual machines can transparently call `Service`s (similar to pods) without requiring any additional configuration.
```

**Review by nrjpoddar — COMMENTED** [review]
For the "Cluster" boundary in your diagrams, can you call them "Kubernetes cluster"?

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:29]: Is this an ingress/egress GW? Does this matter which one?

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:30]:
```suggestion
This ensures that the virtual machines have a stable address to connect to when they are bootstrapping. All other communication between pods and
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:70]:
```suggestion
When adding a virtual machine workload to the mesh, you will need to create a `WorkloadGroup`, that acts as template for each `WorkloadEntry` instance:
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:116]:
```suggestion
For example, for a `Service` named `product` that was composed of a `Pod` and a `WorkloadEntry`:
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (howardjohn) [reply content/en/docs/ops/deployment/vm-architecture/index.md:17]: Basically, that K8s gives a bunch of functionality for free. When you are on a VM, you are missing all of that (dns, health checks, etc). With Istio, you get those functionalities added to VMs *and* the standard istio value add (mtls, telemetry, etc).

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (howardjohn) [reply content/en/docs/ops/deployment/vm-architecture/index.md:29]: It can be any gateway with istiod exposed. It doesn't even need to be an istio gateway technically. Our docs recommend a dedicated gateway.

**Review by howardjohn — COMMENTED** [review]
Thanks @nrjpoddar !!

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (howardjohn) [inline content/en/docs/ops/deployment/vm-architecture/index.md:31]:
```suggestion
This ensures that the virtual machines have a stable address to connect to when they are bootstrapping. Pods and virtual machines can communicate directly with each other without requiring any intermediate Gateway.
```

**Comment by howardjohn** [comment]
> For the "Cluster" boundary in your diagrams, can you call them "Kubernetes cluster"?

Yep. If its alright, I will wait until we are ready to merge this in case there are other comments- updating the diagrams takes a bit more time than the text

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:120]:
```suggestion
With this configuration, requests to `product` would be load-balanced to both the pod and virtual machine workload instances.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (ericvn) [inline content/en/docs/ops/deployment/vm-architecture/index.md:4]:
```suggestion
weight: 25
```

Maybe its 15 so it's placed between Architecture and Deployment Models (already has a 20). Or maybe since you want them to be read first, make it 25. The current placement in the netlify shows it after Deployment models.

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (ericvn) [inline content/en/docs/ops/deployment/vm-architecture/index.md:15]:
```suggestion
This enables legacy applications, or applications not suitable to run in a containerized environment, to get all the benefits that Istio provides to applications running inside Kubernetes.
```

You could remove all and leave same.

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (ericvn) [inline content/en/docs/ops/deployment/vm-architecture/index.md:63]:
```suggestion
* [`WorkloadGroup`](/docs/reference/config/networking/workload-group/) represents a logical group of virtual machine workloads that share common properties. This is similar to a `Deployment` in Kubernetes.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (ericvn) [inline content/en/docs/ops/deployment/vm-architecture/index.md:64]:
```suggestion
* [`WorkloadEntry`](/docs/reference/config/networking/workload-entry/) represents a single instance of a virtual machine workload. This is similar to a `Pod` in Kubernetes.
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (ericvn) [inline content/en/docs/ops/deployment/vm-architecture/index.md:113]:
```suggestion
For example, a `Service` named `product` is composed of a `Pod` and a `WorkloadEntry`:
```

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (ericvn) [inline content/en/docs/ops/deployment/vm-architecture/index.md:124]:
```suggestion
Kubernetes provides DNS resolution in pods for `Service` names allowing pods to easily communicate with one another by stable hostnames.
```

**Comment by ericvn** [comment]
LGTM. If @nrjpoddar approves (I think the claBot has issues that may need to be resolved as well), I'll approve and we can get this in.

**Comment by howardjohn** [comment]
@nrjpoddar any outstanding concerns? thanks for the feedback so far!

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [inline content/en/docs/ops/deployment/vm-architecture/index.md:17]: Here's my suggested rewording:

For workloads running on Kubernetes, the Kubernetes platform itself provides various functionalities like service discovery, DNS resolution, and health checks which are often missing in virtual machine environments. Istio enables these features for workloads running on virtual machines, and in addition allows these workloads to utillize Istio functionalities like mutual TLS (mTLS), rich telemetry and, advanced traffic management capabilities.

**Review by nrjpoddar — APPROVED** [review]
Added one more suggestion, rest LGTM.

**Comment by nrjpoddar** [comment]
@howardjohn can you add "Kubernetes" in front of "cluster" in your diagrams before merging?

File: content/en/docs/ops/deployment/vm-architecture/index.md
Comment (nrjpoddar) [reply content/en/docs/ops/deployment/vm-architecture/index.md]: I still  see them after deployment models in rendered docs.

**Comment by howardjohn** [comment]
> can you add "Kubernetes" in front of "cluster" in your diagrams before merging? Additionally, can you link to this new doc from docs/setup/install/virtual-machine/ page for visibility?

Done, thanks!

**Review by nmittler — APPROVED** [review]

**Pull request merged**
